acceptance/suites/tests/555-puppetserver-service/reboot.rb

```diff
@@ -0,0 +1,55 @@
+if (options[:type] == 'pe') then
+  pssn='pe-puppetserver'
+else
+  pssn='puppetserver'
+end
+
+test_name "Validates the ability of the #{pssn} service to reliably start after a reboot."
+
+max=3
+sleeptime=45
+retryCounter=0
+step "\n-----------\nValidate #{pssn} is up and running, then reboot, repeat #{max} times\n-----------"
+
+for i in 1..max 
+  step "\n-----\nLoop #{i}\n-----"
+   
+  step "check uptime"
+  origUptime=on(master, "awk '{print $1}' /proc/uptime").stdout.chomp.to_i
+
+  step "reboot"
+  #rebooting with on(master,'reboot') causes an IOError: stream end.  This seems like a sane workaround.
+  #QENG-1914 for more...
+  system "ssh -i #{options[:keyfile]} -o StrictHostKeyChecking=no root@#{master.reachable_name} \'reboot\'"
```

Comment:
FWIW, this type of problem looks really difficult to solve with a single command line because shell escaping is getting a bit insane, but this approach seems like it could work.  It's basically a matter of copying a script to the SUT, then kicking off that script as a background process.  The script will delay a moment or two, giving beaker time to gracefully log out, then the script will bounce the box.

Here's the script I played around with:

```sh
! /bin/bash
#
# Delayed reboot that can be triggered from an SSH connection without
# unmercifully killing that very connection.

echo "Kicked off at `date`" >> /tmp/reboot.log
sleep 2
echo "Rebooted at `date`" >> /tmp/reboot.log
# shutdown -r now
```

In the real world the echo statements could be removed, so the script does very little.  It does give us a convenient thing to pass into a bash / nohup / redirection / background job mess of a command:

```
ssh jeff@localhost bash -c "'nohup /tmp/delayed_reboot > /dev/null 2>&1 &'"
```

Note, the single quotes wrapped in double quotes are important so bash, on the other end of the wire, gets the full command expression as a single argument.  This may not be an issue when calling from beaker which isn't subject to shell shell expansion.